server/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -380,6 +384,43 @@ protected QueryBuilder doIndexMetadataRewrite(QueryRewriteContext context) throw
         return this;
     }
 
+    @Override
+    protected QueryBuilder doCoordinatorRewrite(CoordinatorRewriteContext coordinatorRewriteContext) {
+        // if the query (can and) will be rewritten to a term query, we can try and do a coordinator rewrite based on constant fields
+        if (fuzziness != null || lenient) {
+            // Term queries can be neither fuzzy nor lenient, so don't rewrite under these conditions
+            return this;
+        }
+        // If we're using a keyword analyzer then we can rewrite this to a TermQueryBuilder
+        // and possibly shortcut
+        NamedAnalyzer configuredAnalyzer = configuredAnalyzer(coordinatorRewriteContext);
+        if (configuredAnalyzer != null && configuredAnalyzer.analyzer() instanceof KeywordAnalyzer) {
```

Comment:
I don't think we can reliably do this for match query in the coord node. The analyzer bits come from mappings (see configuredAnalyzer) that are not necessarily available on the coord node. There is one case where the query may specify an analyzer, but still that analyzer needs to be looked up via mapper service which is going to be null on the coord node if no shards for the queried index are allocated on it. I think that there is no way around it.

Reply:
Ah great catch. Removing now